storage/range.go

```diff
@@ -1370,3 +1402,119 @@ func (r *Range) AdminSplit(args *proto.AdminSplitRequest, reply *proto.AdminSpli
 		reply.SetGoError(util.Errorf("split at key %q failed: %s", splitKey, err))
 	}
 }
+
+// InitialState implements the raft.Storage interface.
+func (r *Range) InitialState() (raftpb.HardState, raftpb.ConfState, error) {
+	data, err := r.rm.Engine().Get(
```

Comment:
You definitely don't want to just cast to an EncodedKey here, as this will screw up iterations using MVCC.

We store everything via MVCC, just using proto.ZeroTimestamp in order to not encode multiple rows indicating timestamped value.

So, to store a non-MVCC value like the raft state, look to how we store stats for example, in storage/engine/stat.go:

    if err := MVCCPut(engine, nil, MakeRangeStatKey(raftID, stat), proto.ZeroTimestamp, value, nil); err != nil {
      return err
    }